packages/cli-utils/src/index.ts

```diff
@@ -155,8 +138,95 @@ async function main() {
+      const config = await getConfig();
+      if (!config) {
+        return;
+      }
+      const result = (await check(config, opts.severity || 'error')) || [];
+      const errorDiagnostics = result.filter((d) => d.severity === 'error');
+      const warnDiagnostics = result.filter((d) => d.severity === 'warn');
+      const infoDiagnostics = result.filter((d) => d.severity === 'info');
+      if (
+        errorDiagnostics.length === 0 &&
+        warnDiagnostics.length === 0 &&
+        infoDiagnostics.length === 0
+      ) {
+        // eslint-disable-next-line no-console
+        console.log('No issues found! 🎉');
+        return;
+      } else {
+        const errorReport = errorDiagnostics.length
+          ? `Found ${errorDiagnostics.length} Errors:\n\n${constructDiagnosticsPerFile(
+              errorDiagnostics
+            )}\n\n`
+          : ``;
+        const warningsReport = warnDiagnostics.length
+          ? `Found ${warnDiagnostics.length} Warnings:\n\n${constructDiagnosticsPerFile(
+              warnDiagnostics
+            )}\n\n`
+          : ``;
+        const suggestionsReport = infoDiagnostics.length
+          ? `Found ${infoDiagnostics.length} Suggestions:\n\n${constructDiagnosticsPerFile(
+              infoDiagnostics
+            )}\n\n`
+          : ``;
+        // eslint-disable-next-line no-console
+        console.log(`${errorReport}${warningsReport}${suggestionsReport}`);
+      }
     });
   prog.parse(process.argv);
 }
 
+function constructDiagnosticsPerFile(diagnostics: FormattedDisplayableDiagnostic[]): string {
+  const diagnosticsByFile = diagnostics.reduce<Record<string, string[]>>((acc, diag) => {
+    const file = diag.file || '';
+    if (!acc[file]) {
+      acc[file] = [];
+    }
+    acc[file].push(`[${diag.line}:${diag.col}] ${diag.message}`);
+    return acc;
+  }, {});
+
+  return Object.entries(diagnosticsByFile).reduce((acc, [fileName, diagnostics]) => {
+    return `${acc}${fileName}\n${diagnostics.join('\n')}\n\n`;
+  }, '');
+}
+
+const getConfig = async (): Promise<GraphQLSPConfig | undefined> => {
+  const cwd = process.cwd();
+  const tsconfigpath = path.resolve(cwd, 'tsconfig.json');
+
+  // TODO: Remove redundant read and move tsconfig.json handling to internal package
+  const root = (await resolveTypeScriptRootDir(tsconfigpath)) || cwd;
+
+  let tsconfigContents: string;
+  try {
+    const file = path.resolve(root, 'tsconfig.json');
+    tsconfigContents = await fs.readFile(file, 'utf-8');
+  } catch (error) {
+    console.error('Failed to read tsconfig.json in current working directory.', error);
+    return;
+  }
+
+  let tsConfig: TsConfigJson;
+  try {
+    tsConfig = parse(tsconfigContents) as TsConfigJson;
+  } catch (err) {
+    console.error(err);
+    return;
+  }
+
+  const config = getGraphQLSPConfig(tsConfig);
+  if (!config) {
+    console.error(`Could not find a "@0no-co/graphqlsp" plugin in your tsconfig.`);
+    return;
+  }
+
+  return config;
+};
```

Comment:
(doesn't block merging, can be a todo)

We'll have to start pulling the actions out into separate files. 

On a side-note, I'm not a huge fan of `sade` because it doesn't encourage splitting up sub-commands and the main commands (i.e. nested command levels with separate options being co-located with the action)

But anyway, main point, let's already start pulling action functions out

**Strong suggestion to not do this in this PR to not waste time merging this**